### ISSUE TITLE
ProtectBranchAsync throws exception

### ISSUE THREAD

**Issue by nicholasyin** [nmklotas/GitLabApiClient#123]
![image](https://user-images.githubusercontent.com/17007651/76166685-58735500-619b-11ea-9de7-5373d7b475c6.png)
Here's my code
`ProtectBranchAsync(
                            project,
                            new ProtectBranchRequest(
                                "master",
                                ProtectedRefAccessLevels.MAINTAINER_ACCESS,
                                ProtectedRefAccessLevels.MAINTAINER_ACCESS,
                                ProtectedRefAccessLevels.MAINTAINER_ACCESS))`

**Comment by nicholasyin** [comment]
Looks like no one is tracking this issue list. pity...

**Comment by Jetersen** [comment]
There is but equally there is a no expectation that your issue will be resolved without contributing.
At the moment I started at a new workplace where we do not use gitlab. So my interest in the project is only there for my personal interests in open source however I am maintaining several projects.

I would happily accept a full request.

**Comment by nicholasyin** [comment]
> There is but equally there is a no expectation that your issue will be resolved without contributing.
> At the moment I started at a new workplace where we do not use gitlab. So my interest in the project is only there for my personal interests in open source however I am maintaining several projects.
> 
> I would happily accept a pull request.

In another words, "This is my personal project now. It's you guys who decide to download and use, so don't expect anything more than that.". Fair enough...

**Comment by Jetersen** [comment]
It is all about expectations and having unreasonable exception towards open source as someone who does not contribute to open source, I can understand your misconception towards open source as it take time and effort to invest in something.

This was not even my project to begin with, I found issues and started contributing.
I solved the problems I had. I invested my time and energy into the project. I help move the project forward so it should be very easy for you to figure out why by even writing an integration test.
